src/easylink/step.py

```diff
@@ -181,15 +187,23 @@ def update_implementation_graph(
         self.remap_slots(graph, step_config)
         for node in self.graph.nodes:
             step = self.graph.nodes[node]["step"]
-            step.update_implementation_graph(graph, step_config)
+            sub_config = step_config
+            if step.step_key:
+                sub_config = step_config[step.step_key]
+            step.update_implementation_graph(graph, sub_config)
             graph.remove_node(node)
 
     def validate_step(self, step_config: LayeredConfigTree) -> Dict[str, List[str]]:
         """Validate each step in the subgraph in turn. Also return errors for any extra steps."""
         errors = {}
         for node in self.graph.nodes:
             step = self.graph.nodes[node]["step"]
-            step_errors = step.validate_step(step_config)
+            if not step.step_key:
+                step_errors = step.validate_step(step_config)
```

Comment:
If you explicitly implement `validate_step` in `PipelinSchema` you shouldn't need this if branch.